integration/spark2/src/test/scala/org/apache/spark/sql/CarbonAnalysisTest.scala

```diff
@@ -0,0 +1,25 @@
+package org.apache.spark.sql
+
+import org.apache.spark.sql.common.util.Spark2QueryTest
+import org.scalatest.BeforeAndAfterAll
+
+class CarbonAnalysisTest extends Spark2QueryTest with BeforeAndAfterAll {
+
+  override def afterAll(): Unit = {
+    sql("DROP TABLE IF EXISTS table1")
+    sql("DROP TABLE IF EXISTS table2")
+  }
+
+  test("test carbon insert and select table column not equal") {
+    sql("DROP TABLE IF EXISTS table1")
+    sql("DROP TABLE IF EXISTS table2")
+    sql("create table table1 (col1 string, col2 string) partitioned by(pt string) stored by 'carbondata'")
+    sql("create table table2 (t2_c1 string, t2_c2 string, t2_c3 string) partitioned by(pt string)")
+
+    sql("insert overwrite table table2 partition(pt=20200101) values('v11', 'v12', 'v13')")
+    val e = intercept[Exception] {
+      sql("insert into table1 select * from table2")
+    }
+    assert(e.getMessage.contains("number of columns are different"))
+  }
```

Comment:
You mean in this case, user should use `insert into table1 select t2_c1, t2_c2, pt from table2` after adding new column in table2?

Reply:
User should also add column in table1 after adding new column in table2. If not, we should throw exception, like parquet table.

Reply:
ok, agreed